pkg/history/apistore/store.go

```diff
@@ -31,6 +32,9 @@ type Store struct {
 var _ history.Store = (*Store)(nil)
 
 func (s *Store) Append(ctx context.Context, payload []byte) (history.Record, error) {
+	if payload == nil {
+		return history.Record{}, fmt.Errorf("payload cannot be nil")
+	}
```

Comment:
I think this is solving the issue in the wrong place.
Firstly, checking `slice == nil` is bad practice, `len(slice) == 0` is better.
Secondly, an empty slice is a valid payload, it's a serialised proto that has no non-default, non-empty fields.

If we must check a slice for nil then we must comment why we're doing this and why nil is treated differently than an empty slice.